Base/Python/slicer/translation.py

```diff
@@ -0,0 +1,27 @@
+import os
+import inspect
+
+
+def get_context(source_file):
+    """ get the translation context based on the source file name"""
+    if os.path.isfile(source_file):
+        parent_folder = os.path.dirname(source_file)
+        init_file_path = parent_folder + os.path.sep + '__init__.py'
+
+        if os.path.isfile(init_file_path):
+            context_name = os.path.basename(parent_folder)
+            context_name += '.' + os.path.basename(source_file).replace('.py', '')
+            return context_name
+        else:
+            return os.path.basename(source_file).replace('.py', '')
+    else:
+        return os.path.basename(source_file)
+
+
+def _(text):
+    """translation function for python scripted modules"""
+    from slicer import app
+
+    filename = inspect.stack()[1][1]
```

Comment:
I don't think it would be ever time-critical. It is for user-displayable string.
Anything in Python is not really time-critical anyway (you would not implement a hot loop in Python). If you must do it for some reason then you can easily moved the translated string into a member variable.